subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -1113,13 +1114,20 @@ func getFewestOpporPhysicalCell(cl CellList, suggestedNodeSet common.Set) *Physi
 				fewestOpporNumSuggested = numOppor
 				fewestOpporSuggested = pc
 			}
+			if numOppor > 0 {
+				preemptibleCells.Add(pc)
+			}
 		}
 	}
-	if fewestOpporSuggested == nil {
-		return fewestOpporCell
-	} else {
+	if fewestOpporSuggested != nil {
 		return fewestOpporSuggested
 	}
+	if !preemptibleCells.IsEmpty() {
+		for c := range preemptibleCells.Items() {
+			return c.(*PhysicalCell)
+		}
```

Comment:
Return the random one? Do not leverage map's rand.

Reply:
And add comment for why we do this

---
In reply to: [389731936](https://github.com/microsoft/pai/pull/4268#discussion_r389731936) [](ancestors = 389731936)